settings.gradle

```diff
@@ -1,3 +1,23 @@
+pluginManagement {
+    resolutionStrategy {
+        eachPlugin {
+            // Work around https://github.com/gradle/gradle/issues/1697.
+            if (requested.version == null) {
+                def versionPropertyName = (requested.id.id == 'org.jetbrains.kotlin.jvm') ?
+                        "kotlinVersion" : "${requested.id.name}Version"
+                println("Checking for plugin version property '$versionPropertyName'.")
+                if (gradle.rootProject.hasProperty(versionPropertyName)) {
+                    def version = gradle.rootProject.properties[versionPropertyName]
+                    println("Setting '${requested.id.id}' plugin version to $version.")
+                    useVersion version
+                } else {
+                    println("Property '$versionPropertyName' does not exist.")
```

Comment:
..and `logger.error("...")` here?

Reply:
It's not necessarily and error, I believe. Some other magic might kick in to create a default version, and if not, I'd prefer to leave the erroring to Gradle itself, which would then fail the build.

Reply:
Maybe a warning then?

Reply:
Done.